src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs

```diff
@@ -48,14 +50,23 @@ public SearchParameterDefinitionManager(IModelInfoProvider modelInfoProvider)
 
         public Task StartAsync(CancellationToken cancellationToken)
         {
-            var builder = new SearchParameterDefinitionBuilder(
-                _modelInfoProvider,
-                "search-parameters.json");
+            // This method is idempotent because dependent Start methods are not guaranteed to be executed in order.
+            if (!_started)
+            {
+                var bundle = SearchParameterDefinitionBuilder.ReadEmbeddedSearchParameters("search-parameters.json", _modelInfoProvider);
+
+                SearchParameterDefinitionBuilder.Build(
+                    bundle,
+                    UrlLookup,
+                    TypeLookup,
+                    _modelInfoProvider);
 
-            builder.Build();
+                List<string> list = UrlLookup.Values.Where(p => p.Type == ValueSets.SearchParamType.Composite).Select(p => string.Join("|", p.Component.Select(c => UrlLookup[c.DefinitionUrl].Type))).Distinct().ToList();
 
-            _typeLookup = builder.ResourceTypeDictionary;
-            UrlLookup = builder.UriDictionary;
+                CalculateSearchParameterHashAsync();
+
+                _started = true;
+            }
```

Comment:
I think some old changes that have been removed may have sneaked back in here. I don't think we need the `_started` flag or the url lookup values list anymore.

Reply:
Removed _started flag and unnecessary list.